src/main/kotlin/net/maizegenetics/phgv2/cli/AgcCompress.kt

```diff
@@ -0,0 +1,228 @@
+        val agcFileOut = "${dbPath}/assemblies_tmp.agc"
+        val builder = ProcessBuilder()
+        var redirectOutput = tempDir + "/agc_create_output.log"
+        var redirectError = tempDir + "/agc_create_error.log"
+        when (loadOption) {
+            "create" -> {
+                builder.command("conda","run","-n","phgv2-conda","agc","create","-i",fastaFiles,"-o",agcFile,refFasta)
+            }
+            "append" -> {
+                builder.command("conda","run","-n","phgv2-conda","agc","append","-i",fastaFiles,agcFile,"-o",agcFileOut)
+                redirectOutput = tempDir + "/agc_append_output.log"
+                redirectError = tempDir + "/agc_append_error.log"
+            }
+            else -> {
+                println("Invalid load option: ${loadOption}")
+                return false
+            }
+        }
+
+        builder.redirectOutput( File(redirectOutput))
+        builder.redirectError( File(redirectError))
+
+        println("begin Command to create/append:" + builder.command().stream().collect(Collectors.joining(" ")))
+        try {
+            var process = builder.start()
+            var error = process.waitFor()
+            if (error != 0) {
+                println("agc create run via ProcessBuilder returned error code $error")
+                throw IllegalArgumentException("Error running ProcessBuilder to compress agc files: ${error}")
+            }
+            // If this was an "append" command, the file was written to assemblies_tmp.agc
+            // In this case, we need to move the original assemblies.agc file to assemblies_backup_<date>.agc
+            // can then move the assemblies_tmp.agc to assemblies.agc
+
+            if (loadOption == "append") {
+                // Use Files.move instead of File.renameTo because File.renameTo does not work across file systems
+                // and Files.move is platform independent
+                val agcFileBackupFile = Paths.get("${dbPath}/assemblies_backup_${LocalDate.now()}.agc")
+                val agcFileOutFile = Paths.get(agcFileOut)
+
+                val agcFileOrig = Paths.get(agcFile)
+                Files.move(agcFileOrig,agcFileBackupFile)
+                Files.move(agcFileOutFile,agcFileOrig)
+            }
+
+        } catch (exc: Exception) {
+            myLogger.error("Error: could not create agc compressed file.")
+            throw IllegalArgumentException("Error running ProcessBuilder for agc create or append: ${exc.message}")
+        }
+
+        return true
+    }
+
+    // This function takes a list of fasta file names and returns a list of sample names
+    // Sample names are the file names without extension
+    fun getCurrentSampleNames(fastaFiles:List<String>):List<String> {
+        //create a list of names from the fasta files, where the name is
+        // just the file name, no path, and without the extension
+        val sampleNames = mutableListOf<String>()
```

Comment:
Consider:
`return fastaNames.map{ it.substringAfterLast("/").substringBefore(".") }`

Reply:
I think you're trying to prove you have better coding skills than co-pilot!  
I've changed from the "split" commands to using your "substringAfterLast/substringBefore" example.

Reply:
Not sure about that, but the substringBefore and AfterLast tends to read a bit better which should help in the future.